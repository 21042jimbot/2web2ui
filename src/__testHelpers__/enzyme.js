export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
