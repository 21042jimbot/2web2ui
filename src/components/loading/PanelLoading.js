import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Loading } from 'src/components/loading';
import { Panel } from 'src/components/matchbox';
import styles from './PanelLoading.module.scss';

const StyledPanel = styled.div`
  min-height: ${props => props.minHeight};
`;

const PanelLoading = props => {
  const { minHeight, accent, title } = props;

  return (
    <StyledPanel
      as={Panel.LEGACY}
      className={styles.Loading}
      accent={accent}
      data-id="panel-loading"
      title={title}
    >
      <Loading minHeight={minHeight} />
    </StyledPanel>
  );
};

PanelLoading.propTypes = {
  minHeight: PropTypes.string,
  accent: PropTypes.bool,
};

PanelLoading.defaultProps = {
  minHeight: '400px',
  accent: false,
};

export default PanelLoading;
