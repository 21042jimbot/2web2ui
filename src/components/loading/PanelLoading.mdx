# Panel Loading

Use the `<PanelLoading/>` component as a placeholder for `<Panel/>` component. This is particularly
useful within `<Modal/>` components.

## Usage

```js
import React from 'react';
import { PanelLoading } from 'src/components/loading';
import { Panel } from 'src/components/matchbox';

const MyComponent = ({ loading }) => (loading ? <PanelLoading /> : <Panel />);
```
