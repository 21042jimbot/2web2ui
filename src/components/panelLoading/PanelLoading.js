import React from 'react';
import styles from './PanelLoading.module.scss';
import { Panel } from '@sparkpost/matchbox';
import { Loading } from 'src/components/loading/Loading';

const PanelLoading = ({ minHeight }) => (
  <Panel className={styles.Loading} style={{ minHeight }}>
    <Loading />
  </Panel>
);

PanelLoading.defaultProps = {
  minHeight: '400px'
};

export default PanelLoading;
