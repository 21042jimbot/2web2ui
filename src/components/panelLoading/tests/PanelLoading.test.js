import React from 'react';
import PanelLoading from '../PanelLoading';
import { render } from 'enzyme';

describe('Panel Loading', () => {
  test('render', () => {
    const wrapper = render(<PanelLoading />);

    expect(wrapper).toMatchSnapshot();
  });
});
