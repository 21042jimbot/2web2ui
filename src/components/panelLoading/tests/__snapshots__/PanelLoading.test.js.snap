// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Loading render 1`] = `
<Panel
  style={
    Object {
      "minHeight": "400px",
    }
  }
>
  <Loading />
</Panel>
`;
